janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -6165,4 +6165,234 @@ private void testWritingAndReading(File f) {
+                option(KCVSLog.LOG_READ_LAG_TIME, MANAGEMENT_LOG), Duration.ofMillis(50),
+                option(LOG_READ_INTERVAL, MANAGEMENT_LOG), Duration.ofMillis(250)
+        );
+    	
+        String indexName = "edgesByAssocKindOut";
+
+        //Schema creation
+        initReindexSchema();
+        EdgeLabel edgeLabel = mgmt.getEdgeLabel("egLabel");
+        PropertyKey propAssocKind = mgmt.getPropertyKey("assocKind");
+        mgmt.buildEdgeIndex(edgeLabel, indexName, Direction.OUT, propAssocKind);
+        finishSchema();
+
+        //Create vertex
+        JanusGraphVertex a = tx.addVertex();
+        a.property("vtName","A");
+        JanusGraphVertex b = tx.addVertex();
+        b.property("vtName","B");
+        //add Edge
+        a.addEdge("egLabel",b,"assocKind",1);
+        b.addEdge("egLabel",a,"assocKind",2);
+        newTx();
+
+        RelationType t = mgmt.getRelationType("egLabel");
+        RelationTypeIndex relationIndex = mgmt.getRelationIndex(t,indexName);
+        assertEquals(SchemaStatus.ENABLED, relationIndex.getIndexStatus());
+        
+        //asserting before reindex
+        assertEquals(0,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(1,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(1,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(0,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 2,false));
+        assertEquals(1,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(0,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(1,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 2,false));
+
+        //Reindexing
+        mgmt.updateIndex(relationIndex, SchemaAction.REINDEX).get();
+        finishSchema();
+
+        relationIndex = mgmt.getRelationIndex(t,indexName);
+        assertEquals(SchemaStatus.ENABLED, relationIndex.getIndexStatus());
+
+        //asserting after reindex
+        assertEquals(0,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(1,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(1,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(0,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 2,false));
+        assertEquals(1,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(0,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(1,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 2,false));
+    }
+	
+    private void initReindexSchema() {
```

Comment:
Once you extract test logic into a single helper method, e.g. "testReindexingForEdgeIndex(direction)", then you should be able to put the preparation work in that method and dismiss this method.

Reply:
testcase removed as this seems to be unnecessary for this PR